includes/abstracts/abstract-batch.php

```diff
@@ -271,8 +318,20 @@ public function run( $current_step ) {
 		$per_page = apply_filters( 'loco_batch_' . $this->slug . '_per_page', $per_page );
 
 		$total_steps = ceil( $this->total_num_results / $per_page );
+
 		if ( (int) $this->current_step === (int) $total_steps ) {
-			$this->update_status( 'finished' );
+
+			// Need to really check to make sure there were no results added while processing.
+			// In the case of destructive actions (i.e. deletion) there will be a gap equal to the per_page param.
+			// In all other cases, the difference in totals should equal total number of results.
+			// If neither of these are true, we need to run the last step over again.
+			$difference = $this->total_num_results - $this->difference_in_result_totals;
```

Comment:
That difference accounts for the gap between the original total and the most recent query. In the case of a destructive action, the final step will have a number that’s _exactly_ equal to the posts_per_page number. So if there’s 20 total in 4 steps, the last step would have 5 results left.

In all other cases, that difference would just be the same as the total_num_results so we know we are done.

Now why the greater than or equal to? Well, let’s say we have a destructive action, but we also added a post. In the example above then that would increase 20 to 21 and the final step would actually only have 1 example. But it will never have _above_ 5.